### DISCUSSION TITLE
Generate C# code from text input

### DISCUSSION THREAD

**Discussion by PBuchmann** [dotnet/roslyn#50145] [Q&A] [answered]
I have written a T4 generator that generates a complete C# file from a csv file as input and am thinking of rewriting it as a C# source code generator. The csv file contains a rights matrix that needs to be converted to C# code. I could think of a lot of other scenarios where source code generation from text files could be very helpful.

Can such scenarios be implemented with source generators? If so, are there any examples of this?

**Comment by svick** [comment, marked as answer]
Yes, [source generator samples](https://github.com/dotnet/roslyn-sdk/tree/d7d6a76/samples/CSharp/SourceGenerators) have [one for generating code from CSV files](https://github.com/dotnet/roslyn-sdk/blob/d7d6a76/samples/CSharp/SourceGenerators/SourceGeneratorSamples/CsvGenerator.cs). (Though the description hasn't been updated to indicate Source Generators are no longer in preview.)
